crates/ark/src/data_explorer/r_data_explorer.rs

```diff
@@ -379,6 +379,11 @@ impl RDataExplorer {
             if self.sort_keys.len() > 0 {
                 self.sorted_indices = Some(r_task(|| self.r_sort_rows())?);
             }
+
+            // This is a no-op if there are no filters to be applied.
+            let (indices, _) = self.row_filters_compute()?;
+            self.filtered_indices = indices;
+
             self.apply_sorts_and_filters();
```

Comment:
```suggestion
            // Recompute and apply filters and sorts.
            let (indices, _) = self.row_filters_compute()?;
            self.filtered_indices = indices;
            self.apply_sorts_and_filters();
```

Maybe just copy/paste from the `if` branch path? Might help us keep it in sync. (I think the no-op can be implied)